lineman/app/services/modal_service.coffee

```diff
@@ -0,0 +1,25 @@
+angular.module('loomioApp').factory 'ModalService', ($modal, $rootScope) ->
+  new class ModalService
+    openModal: (name, options = {}) ->
+      return console.log "Invalid modal name: #{name}" unless @modals[name]
+      $rootScope.$broadcast 'modalOpened'
+      $modal.open
+        templateUrl: @modals[name].template
+        controller:  @modals[name].controller
+        resolve:     @setLocals name, options
+
+    setLocals: (name, options) ->
+      locals = {}
+      _.each @modals[name].locals, (local) ->
+        locals[local] = options[local]
+      locals
+
+    modals:
```

Comment:
A list of modals with their associated templates and controllers.

This will hopefully replace DiscussionFormService, ProposalFormService, and the myriad of places where we're duplicating modal-calling code.